packages/flutter_tools/lib/src/custom_devices/custom_device.dart

```diff
@@ -594,6 +595,33 @@ class CustomDevice extends Device {
   @override
   Future<bool> get isLocalEmulator async => false;
 
+  @override
+  bool get supportsScreenshot => _config.supportsScreenshotting;
+
+  @override
+  Future<void> takeScreenshot(File outputFile) async {
+    if (supportsScreenshot == false) {
+      throw UnsupportedError('Screenshotting is not supported for this device.');
+    }
+
+    final List<String> interpolated = interpolateCommand(_config.screenshotCommand, <String, String>{});
+
+    final RunResult result = await _processUtils.run(interpolated, throwOnError: true);
+
+    // remove all non-base64 characters.
+    final String trimmedStdout = result.stdout.trim().replaceAll(RegExp(r'[^a-zA-Z0-9+/]+'), '');
```

Comment:
This seems like a bit of an awkward way of communicating the data. If there is non-base64 data output it could mean that the command failed unexpectedly, in which case you might just be creating an "image" out of a garbled string. Seems like failing would be the better choice

Reply:
My thought was that if something goes wrong, `ssh` should exit with an error code, that's why `throwOnError` is true and nothing is trying to catch the exception (don't really know what to do except that, should it just silently fail?)

Though, at least the `|` doesn't pass on failures, so maybe it'll run into the case you mentioned when the `cat` command unexpectedly fails. I'll change it